### ISSUE TITLE
`process = { env: {} }` leaks into global namespace

### ISSUE THREAD

**Issue by TheoBr** [apollographql/invariant-packages#107]
*tl;dr* is that an `exports.process = { env: {} };` call is being made (likely from [here](https://github.com/apollographql/invariant-packages/blob/main/packages/ts-invariant/src/invariant.ts#L70)), which causes a `process.env` variable to appear as seen below.

![image](https://user-images.githubusercontent.com/6751787/113648661-cd84b880-9641-11eb-8355-0a1e30a175d4.png)

This lead to a really rough debug session. Our app uses `musickit.js` from Apple, which failed to identify what type of module to return since `process.env` existed.

We ran into this by installing [pubnub/react](https://github.com/pubnub/react/), but I suspect many others will in the future

Package.json deps in case they help. We're using `vite` and `rollup` :)
```json
{
  "dependencies": {
    "@react-hook/debounce": "^3.0.0",
    "howler": "^2.2.1",
    "jotai": "^0.14.0",
    "pubnub": "^4.29.11",
    "pubnub-react": "^2.1.0",
    "react": "^17.0.0",
    "react-dnd": "^11.1.3",
    "react-dnd-html5-backend": "^11.1.3",
    "react-dom": "^17.0.0",
    "react-double-marquee": "^1.1.0",
    "react-ga": "^3.3.0",
    "react-is": "^17.0.2",
    "react-json-view": "^1.21.1",
    "react-phone-input-2": "^2.14.0",
    "react-query": "^3.8.2",
    "socket.io-client": "^4.0.0",
    "styled-components": "^5.2.3",
    "uuid": "^8.3.2",
    "wouter": "^2.7.3",
    "zustand": "^3.3.3"
  },
  "devDependencies": {
    "@types/howler": "^2.2.2",
    "@types/pubnub": "^4.29.1",
    "@types/react": "^17.0.0",
    "@types/react-dom": "^17.0.0",
    "@types/react-phone-number-input": "^3.0.6",
    "@types/styled-components": "^5.1.7",
    "@types/uuid": "^8.3.0",
    "@typescript-eslint/eslint-plugin": "^4.19.0",
    "@typescript-eslint/parser": "^4.19.0",
    "@vercel/node": "^1.9.1",
    "@vitejs/plugin-react-refresh": "^1.3.1",
    "esbuild": "^0.8.47",
    "eslint": "^7.23.0",
    "eslint-plugin-prettier": "^3.3.1",
    "node-fetch": "^2.6.1",
    "prettier": "2.2.1",
    "typescript": "^4.0.0",
    "vite": "^2.1.5"
  }
}
```

**Comment by glasser** [comment]
While @benjamn is the expert here, I believe this exact issue is fixed in 0.7.0; perhaps pubnub/react is pulling in an older version?

**Comment by TheoBr** [comment]
You appear to be correct - just filed a PR that updates `pubnub/react` to the latest version :)

https://github.com/pubnub/react/pull/67

**Comment by TheoBr** [comment]
Wait nope still leaking after update - worked on local/dev but not in prod :( 
![image](https://user-images.githubusercontent.com/6751787/113669693-f79da100-9668-11eb-9c2a-3dc0ac2c0b46.png)
